department-of-veterans-affairs/abd-vro
# MCP-2113: PDF updates to populate MAS data

## What was the problem?

PDF generator was missing some conditions for populating tables based on [iMVP PDF requirements](https://docs.google.com/document/d/1oSmrpquL98KoNzMAKITcUiCIaG4fXTcSmir0VJwsQsI/edit#). Veteran service location data was not included in claim data model. 

Summary of condition logic that has been added:

1. Claims for increase do not display data that is needed for new claims. (evidence for service connection)
2. In Section IV display relevant conditions only from Lighthouse. 
3. In Section V, 
- display all the conditions from MAS.
- display medication marked as relevant from MAS

Associated tickets or Slack threads:

- [MCP-2113](https://amida.atlassian.net/browse/MCP-2113)

## How does this fix it?[^1]

This PR adds veteran service location data model and conditionals in the PDF generator. Provides a closer adherence as a starting off point between the PDF requirements and the actual data we are getting from MAS in the current state of MAS integration work. Also addresses change required by #821 in sufficiency decision tree. Adds a new 'dateFormatted' parameter to all blood pressures, conditions and medication to display correctly on the PDF.

## How to test this PR
- Step 1 Run app
- Step 2 copy and paste json from `app/src/test/resources/test-data/pdf-generator-mas.json` into PDF generator endpoint and replace the Diagnostic code `7101` with `0000`. Call the other endpoint to get PDF. 
- Step 3 change disability claim type to "INCREASE" and see that evidence for service connection is not rendered
- 
- [^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
